error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java

```diff
@@ -794,6 +794,40 @@ Flux<S> after(Flux<T> flux) {
     }
   }
 
+  /** Prefer {@link Mono#ofType(Class)} over more contrived alternatives. */
+  static final class MonoOfType<T, S> {
+    @BeforeTemplate
+    Mono<S> before(Mono<T> mono) {
+      return mono.filter(
+              Refaster.anyOf(
+                  Refaster.<S>clazz()::isInstance,
+                  x -> Refaster.<S>clazz().isAssignableFrom(x.getClass())))
```

Comment:
We can introduce a separate rule to prefer the first over the second. (I have this change locally already, but need to leave the train now :sweat_smile:.)

Reply:
I filed #811 for this.